### PR TITLE
Remove references to `EncryptedSharedPreferences` architecture overview (COMMUNITY)

### PR DESCRIPTION
**Pull request by fynngodau**
`EncryptedSharedPreferences` are no longer used since version 1.15.x through https://github.com/corona-warn-app/cwa-app-android/pull/2468#issuecomment-787045765. Therefore, we should remove the reference to it being used from the documentation.

### CODE REVIEW

File: docs/architecture-overview.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Licensing: [Apache License 2.0](https://github.com/JodaOrg/joda-time/blob/master
 
 
 ### Room
-Room is a persistence library that provides an abstraction layer over SQLite. In contrary against the EncryptedSharedPreferences Room is used for storing more complex data.
+Room is a persistence library that provides an abstraction layer over SQLite. In contrary against the SharedPreferences Room is used for storing more complex data.
```
Comment (chiljamgossow) [inline docs/architecture-overview.md:55]: Awkward English sentence :)  Maybe we could correct that as well
As opposed to SharedPreferences, Room..... or
Contrary to SharedPreferences, Room...

**Comment by davidlindercwa** [comment]
Please move to 2.16

**Review by CV113 — DISMISSED** [review]
UA approved

File: docs/architecture-overview.md (hunk 1 above)
Comment (fynngodau) [reply docs/architecture-overview.md:55]: Thanks, I felt weird about it too :)

**Review by mtwalli — APPROVED** [review]

**Comment by mtwalli** [comment]
Thank you @fynngodau !

**Review by chiljamgossow — APPROVED** [review]
Nice! Thanks

**Pull request merged**